### PR TITLE
Get the Deployment Type

### PR DESCRIPTION
**Pull request by bhagyagkwd**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPIInstanceDataSource_basic
--- PASS: TestAccIBMPIInstanceDataSource_basic (45.75s)
PASS

=== RUN   TestAccIBMPIInstanceDeploymentType
--- PASS: TestAccIBMPIInstanceDeploymentType (387.58s)
PASS
...
```

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_instance.go [hunk 1]
```diff
@@ -224,6 +228,7 @@ func dataSourceIBMPIInstancesRead(ctx context.Context, d *schema.ResourceData, m
 	d.Set("storage_pool_affinity", powervmdata.StoragePoolAffinity)
 	d.Set("license_repository_capacity", powervmdata.LicenseRepositoryCapacity)
 	d.Set("networks", flattenPvmInstanceNetworks(powervmdata.Networks))
+	d.Set("deployment_type", powervmdata.DeploymentType)
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_instance.go:231]: Please also set the response value in `resource_ibm_pi_instance` read method.

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Comment by hkantare** [comment]
Add the documentation update

**Review by yussufsh — APPROVED** [review]
lgtm

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 2]
```diff
@@ -481,7 +481,7 @@ func resourceIBMPIInstanceRead(ctx context.Context, d *schema.ResourceData, meta
 		d.Set("min_virtual_cores", powervmdata.VirtualCores.Min)
 	}
 	d.Set(helpers.PIInstanceLicenseRepositoryCapacity, powervmdata.LicenseRepositoryCapacity)
-
+	d.Set(PIInstanceDeploymentType, powervmdata.DeploymentType)
```
Comment (hkantare) [inline ibm/service/power/resource_ibm_pi_instance.go:484]: If user don't pass any deploymentType during create what does the GET return back ?

**Comment by hkantare** [comment]
fix the go.mod conflict

**Comment by hkantare** [comment]
fix the travis failure and check the comments

**Comment by hkantare** [comment]
I see only go.mod and sum files ..no other file changes

File: ibm/service/power/resource_ibm_pi_instance.go (hunk 2 above)
Comment (bhagyagkwd) [reply ibm/service/power/resource_ibm_pi_instance.go:484]: returns empty

**Review by rmoralesjr — CHANGES_REQUESTED** [review]
Do you ever read in the deployment type field in the resourceIBMPIInstanceRead function in
ibm/service/power/resource_ibm_pi_instance_test.go ?

**Review by rmoralesjr — APPROVED** [review]
I missed the change was done already to resourceIBMPIInstanceRead

**Pull request merged**